conda.sh

```diff
@@ -57,6 +60,14 @@ generate_numba_environment_yaml()
     grep -Ev "numba|python" environment.yml > environment.numba.yml
 }
 
+generate_ray_environment_yaml()
+{
+    # Limit max Python version and append pip install ray
+    echo "Generating \"environment.ray.yml\" File"
+    ray_python=`./ray_python_version.py`
+    sed "/  - python/ s/$/,<=$ray_python/" environment.yml | cat - <(echo $'  - pip\n  - pip:\n    - ray>=2.23.0') > environment.ray.yml
```

Comment:
Note that I had to add the initial `sed` command to prevent `python` from being upgraded to the latest version and, instead, just keep whatever version already exists. Otherwise, right now, `conda` is forcing my `python` to get upgraded from `3.11` to `3.12` and the latter is incompatible with `ray`